nx/lib/nx.ex

```diff
@@ -1217,6 +1200,25 @@ defmodule Nx do
     raise ArgumentError, "expected a %Nx.Tensor{} or a number, got: #{inspect(t)}"
   end
 
+  defp to_tensor(%T{} = tensor, fun) do
```

Comment:
```suggestion
  defp to_tensor(%T{} = tensor, fun) when is_function(fun, 1) do
```

The same would apply to the other clauses

Reply:
I think it is fine since it is private API although I would definitely add it if it was public API. :)